src/test/java/org/elasticsearch/index/engine/internal/InternalEngineTests.java

```diff
@@ -1559,4 +1562,78 @@ public void testSettings() {
 
         }
     }
+
+    @Test
+    public void testRetryWithAutogeneratedIdWorksAndNoDuplicateDocs() throws IOException {
+
+        ParsedDocument doc = testParsedDocument("1", "1", "test", null, -1, -1, testDocument(), B_1, false);
+        boolean canHaveDuplicates = false;
```

Comment:
This shouldn't have any effect any more, right? do we want to randomize it with a comment it's being removed?